Lib/csv.py

```diff
@@ -80,6 +81,8 @@ class unix_dialect(Dialect):
 class DictReader:
     def __init__(self, f, fieldnames=None, restkey=None, restval=None,
                  dialect="excel", *args, **kwds):
+        if fieldnames is not None and not isinstance(fieldnames, Sequence):
+            raise TypeError(f"Expected a sequence, got {type(fieldnames)}")
```

Comment:
If this change doesn't break any tests then some tests are missing.

Reply:
Hopefully resolved in [https://github.com/python/cpython/pull/32225/commits/66134441a25e431bc1e0f43151ba5c83aa9c6fbf](https://github.com/python/cpython/pull/32225/commits/66134441a25e431bc1e0f43151ba5c83aa9c6fbf)?